cloudinit/sources/helpers/azure.py

```diff
@@ -180,12 +180,15 @@ def get_system_info():
     return evt
 
 
-def report_diagnostic_event(str):
+def report_diagnostic_event(
+        msg: str, logger_func=None) -> events.ReportingEvent:
```

Comment:
Thanks for pointing this out @OddBloke. @anhvoms pointed out above that he'd prefer this to be an optional kwarg in case we do not want to log information.

Reply:
Fair enough; I'd like to see us at least make it a keyword-only argument, as that's how we're using it now (and I think it makes the call sites easier to read:

```suggestion
        msg: str, *, logger_func=None) -> events.ReportingEvent:
```